code/game/objects/structures/windoor_assembly.dm

```diff
@@ -261,18 +262,60 @@ obj/structure/windoor_assembly/Destroy()
 
 
 //Rotates the windoor assembly clockwise
+//These directions are fucked up, apparently dm rotates anticlockwise by default
+/obj/structure/windoor_assembly/verb/rotate()
+	set name = "Rotate Windoor Clockwise"
+	set category = "Object"
+	set src in oview(1)
+
+	var/targetdir = turn(src.dir, 270)
+
+	for(var/obj/obstacle in get_turf(src))
+		if (obstacle == src)
+			continue
+
+		if((obstacle.flags & ON_BORDER) && obstacle.dir == targetdir)
+			usr << span("danger", "You can't turn the windoor assembly that way, there's already something there!")
+			return
+
+	if (src.anchored)
+		usr << "It is fastened to the floor; therefore, you can't rotate it!"
+		return 0
+	if(src.state != "01")
+		update_nearby_tiles(need_rebuild=1) //Compel updates before
+
+	src.set_dir(targetdir)
+
+	if(src.state != "01")
+		update_nearby_tiles(need_rebuild=1)
+
+	update_icon()
+	return
+
+
+//Rotates the windoor assembly anticlockwise
 /obj/structure/windoor_assembly/verb/revrotate()
-	set name = "Rotate Windoor Assembly"
+	set name = "Rotate Windoor Anticlockwise"
```

Comment:
same difference. But yours is longer, and this is already a very long name for a verb, so no